src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -16,32 +16,47 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </summary>
     internal partial class Binder
     {
-        /// <summary>
-        /// Only handles assignment-only or declaration-only deconstructions at this point.
-        /// Issue https://github.com/dotnet/roslyn/issues/15050 tracks allowing mixed deconstructions
-        /// </summary>
         private BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, DiagnosticBag diagnostics)
         {
             var left = node.Left;
             var right = node.Right;
+            CSharpSyntaxNode declaration = null;
```

Comment:
Since `declaration` and `expression` must be either `null` or `node`, could they just be Boolean values (`isDeclaration` and `isExpression`)?